eth/tracers/native/call_flat.go

```diff
@@ -211,6 +211,12 @@ func (t *flatCallTracer) OnTxStart(env *tracing.VMContext, tx *types.Transaction
 }
 
 func (t *flatCallTracer) OnTxEnd(receipt *types.Receipt, err error) {
+	t.ctx = &tracers.Context{
+		BlockHash:   receipt.BlockHash,
```

Comment:
Here you have to be careful: first important to check the error first. Receipt will be nil in that case.
Second: we have some callsites where receipt only has the gasUsed filled in, e.g. https://github.com/ethereum/go-ethereum/pull/30332